src/main/java/uk/gov/hmcts/reform/sscs/services/mapper/CaseDataEventBuilder.java

```diff
@@ -0,0 +1,131 @@
+            return appealCase.getMinorStatus().stream()
+                .filter(minorStatus -> areConditionsToCreatePostponedEventMet(minorStatus.getStatusId(), appealCase))
+                .filter(minorStatus -> postponedEventIsNotPresentAlready(minorStatus.getDateSet(),
+                    appealCase.getMajorStatus()))
+                .map(minorStatus -> buildNewPostponedEvent(minorStatus.getDateSet()))
+                .distinct()
+                .collect(Collectors.toList());
+        }
+        return Collections.emptyList();
+    }
+
+    private boolean areConditionsToCreatePostponedEventMet(String statusId, AppealCase appealCase) {
+        //fixme remove minor status id 26 logic
+        if ("26".equals(statusId)) {
+            return true;
+        }
+
+        if (minorStatusIdIs27AndThereIsOnlyOnePostponementRequest(statusId, appealCase)) {
+            return true;
+        }
+
+        if (minorStatusIdIs27AndMoreThanOnePostponementRequest(statusId, appealCase)) {
+            return !appealCase.getPostponementRequests().stream()
+                .filter(postponementRequest -> "Y".equals(postponementRequest.getPostponementGranted()))
+                .filter(postponementRequest -> postponementRequest.getAppealHearingId().equals(
+                    appealCase.getHearing().get(0).getHearingId()))
+                .collect(Collectors.toList())
+                .isEmpty();
+        }
+        return false;
+    }
+
+    private boolean minorStatusIdIs27AndMoreThanOnePostponementRequest(String statusId, AppealCase appealCase) {
+        return "27".equals(statusId) && appealCase.getPostponementRequests() != null
+            && !appealCase.getPostponementRequests().isEmpty();
+    }
+
+    private boolean minorStatusIdIs27AndThereIsOnlyOnePostponementRequest(String statusId, AppealCase appealCase) {
+        return "27".equals(statusId) && appealCase.getPostponementRequests() != null
+            && !appealCase.getPostponementRequests().isEmpty()
+            && appealCase.getPostponementRequests().size() == 1
+            && "Y".equals(appealCase.getPostponementRequests().get(0).getPostponementGranted());
+    }
+
+    private Events buildNewPostponedEvent(ZonedDateTime dateSet) {
+        return Events.builder()
+            .value(Event.builder()
+                .type(GapsEvent.HEARING_POSTPONED.getType())
+                .date(dateSet.toLocalDateTime().toString())
+                .description(GapsEvent.HEARING_POSTPONED.getDescription())
+                .build())
+            .build();
+    }
+
+    private boolean postponedEventIsNotPresentAlready(ZonedDateTime minorStatusDate,
+                                                      List<MajorStatus> majorStatusList) {
+        return majorStatusList
+            .stream()
+            .filter(majorStatus -> majorStatus.getStatusId().equals(GapsEvent.HEARING_POSTPONED.getStatus()))
+            .filter(majorStatus -> majorStatus.getDateSet().equals(minorStatusDate))
```

Comment:
There is no major status for postponement. So this condition is not valid.

Reply:
removed whole method, it does not make sense.